jenkins/vs/buildimage-vs-image-pr/Jenkinsfile

```diff
@@ -10,6 +10,7 @@ pipeline {
         stage('Prepare') {
             steps {
                 dir('sonic-buildimage') {
+                    sh 'git submodule foreach --recursive "git reset --hard"'
```

Comment:
this is before the checkout, what if the directory is still empty, will this fail?

Reply:
I too thought so, but the previous line (line 12) does cd sonic-buidimage. I guess this jenkinsfile assumes that the buildimage repo always exists. Other jenkins files (broadcom, baseimage etc) dont have this `dir('sonic-buildimage')` step.

To be safe, can we add `|| true`  suffix to that line?
Like:
`sh 'git submodule foreach --recursive "git reset --hard" || true'`

Reply:
Wonder, why not we just cleanup like in brcm-all ? We know, it works.
step([$class: 'WsCleanup'])

Reply:
Because of `dir('sonic-buildimage')` step.
Other jenkins scripts dont have it, so I guess they can delete whole workspace and start from scratch.

Reply:
we should probably put th wscleanup directly under the steps.

Reply:
@lguohan, I did not do that because I cannot verify it.. So, went for the safest approach.

Reply:
Why not we mimic as in vs/buildimage-vs-all-pr/Jenkinsfile. Here it does proper cleanup.

Reply:
Even broadcom/buildimage-brcm-all-pr/Jenkinsfile has the same:
```
        stage('Prepare') {
            steps {
                step([$class: 'WsCleanup'])
               checkout([$class: 'GitSCM',
...

```

Skip  dir('sonic-buildimage') block and do it as in "jenkins/broadcom/buildimage-brcm-all-pr/Jenkinsfile"

Reply:
I did not want to change the basic flow of this jenkins file because I cannot test these changes.. There are other references to `cd sonic-buildimage` in this jenkinsfile. Not sure if the changes you are asking can affect those steps. So I took the safer option of adding a `git reset` on top of existing steps.

Reply:
This change from WsCleanup is brought by the commit https://github.com/Azure/sonic-build-tools/commit/e52facfb9587508c42322919b3f807303f289a6b

@lguohan, can you please confirm, if we can add `wscleanup` directly under the steps, before the `dir('sonic-buildimage')` line